### ISSUE TITLE
[Custom rutimes - Raster] Mosaic Builder UI

### ISSUE THREAD

**Issue by vincentsarago** [developmentseed/eoAPI#59]
Let's create a simple Demo UI to create Mosaic

in titiler-pgstac (eoapi.raster), a Mosaic is a stac-api (pgSTAC) Search request, which we register through the `/mosaic/register` endpoint (https://stac-utils.github.io/titiler-pgstac/advanced/metadata/) 

### Requirements:

- Per Collection Mosaic: Ideally a mosaic is `Per Collection` (because then you can assume that all items have the same assets or at least share a minimal set of assets)
- add bounds for the mosaic from the Collection metadata 
- add set of `defaults` mosaic options (asset selection, expression, color correction, rescaling ....) from which we can construct Tile URLs
- add an option to get Min/Max zoom from all the assets (or selected asset 🤷) 

Note: To be able to fully create a `perfect` mosaic, the user might need to get statistics/info from items so it will be nice to have a place for this.

**Comment by vincentsarago** [comment]
I've added a first version with https://github.com/developmentseed/eoAPI/pull/106 but some features still need to be added

**Comment by vincentsarago** [comment]
## To Do

1. get more assets metadata

within the current builder we get the list of assets from the collection metadata https://github.com/developmentseed/eoAPI/blob/7b429e8106438faf5f4cb24cd09bb23b7edb6b80/runtime/eoapi/raster/eoapi/raster/templates/mosaic-builder.html#L450-L482

This is nice but not enough, we should add a button to retrieve information about Item (using `STACReader.info()` for the first item in a collection, assuming all items are similar 😬 🤷 )

2. add link to viewer 

When we register the mosaic we get a set of links
https://github.com/developmentseed/eoAPI/blob/7b429e8106438faf5f4cb24cd09bb23b7edb6b80/runtime/eoapi/raster/eoapi/raster/templates/mosaic-builder.html#L588-L595

blocked by https://github.com/stac-utils/titiler-pgstac/issues/114

**Comment by zacharyDez** [comment]
@vincentsarago; isn't this implemented now?

**Comment by vincentsarago** [comment]
@zacharyDez some of it yes but it still needs some work to be ✨
